protoc-gen-go/grpc/grpc.go

```diff
@@ -394,8 +394,12 @@ func (g *grpc) generateServerMethod(servName, fullServName string, method *pb.Me
 		g.P("Server: srv,")
 		g.P("FullMethod: ", strconv.Quote(fmt.Sprintf("/%s/%s", fullServName, methName)), ",")
 		g.P("}")
-		g.P("handler := func(ctx ", contextPkg, ".Context, req interface{}) (interface{}, error) {")
-		g.P("return srv.(", servName, "Server).", methName, "(ctx, req.(*", inType, "))")
+		g.P("handler := func(ctx ", contextPkg, ".Context, req interface{}) (reply interface{}, err error) {")
+		g.P("r, err := srv.(", servName, "Server).", methName, "(ctx, req.(*", inType, "))")
+		g.P("if r != nil {")
+		g.P("reply = r")
+		g.P("}")
```

Comment:
Instead of

```
r, err := srv.(GeneratedServer).Method(ctx, req.(*proto.Message))
if r != nil {
       reply = r
}
return
```

explicitly return `nil`

```
if r, err := srv.(GeneratedServer).Method(ctx, req.(*proto.Message)); r != nil {
        return r, err
} else {
        return nil, err
}
```